apps/web/vibes/soul/sections/product-detail/product-detail-form.tsx

```diff
@@ -86,7 +91,26 @@ export function ProductDetailForm<F extends Field>({
     defaultValue,
     shouldValidate: 'onSubmit',
     shouldRevalidate: 'onInput',
+    onSubmit(event, { formData }) {
+      // prevent `formAction` from being called, we'll call `action` ourselves
+      event.preventDefault();
+
+      startTransition(async () => {
+        const payload = await action({ fields, lastResult }, formData);
+
+        console.log(payload);
+
+        if (payload.lastResult?.status === 'success') {
+          toast.success(payload.successMessage);
+        }
+
+        if (payload.lastResult?.status === 'error') {
+          toast.error(payload.lastResult.error?.['']);
```

Comment:
This seems a little off, but not sure if this is the expected pattern when an error message is not tied to a specific id.